File: MaratoneiA1/app/src/main/java/com/example/manoel/maratoneia1/ViewPagerAdapter.java
package com.example.manoel.maratoneia1;

import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import java.util.ArrayList;
import java.util.List;

public class ViewPagerAdapter extends FragmentPagerAdapter {

    //Atributes
    private final List<Fragment> fragmentList = new ArrayList<>();
    private final List<String> fragmentListTitles = new ArrayList<>();


    public ViewPagerAdapter(FragmentManager fm){
        super(fm);
    }


    //Override methods of the class FragmentPagerAdapter
    @Override
    public Fragment getItem(int position) {
        return null;
    }

    //Override the titles of fragments
    @Override
    public CharSequence getPageTitle(int position) {
        return fragmentListTitles.get(position);
    }

    @Override
    public int getCount() {
        return 0;
    }


    //Method that create a new tabItem with a fragment
    public void addFragment(Fragment fragment, String title){
        //Add fragment and titles of tabItems
        fragmentList.add(fragment);
        fragmentListTitles.add(title);

    }
}

File: MaratoneiKotlin/app/src/main/java/com/example/manoeltoj/maratoneikotlin/ViewPagerAdapter.kt
package com.example.manoeltoj.maratoneikotlin

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter

class ViewPagerAdapter(fm: FragmentManager?) : FragmentPagerAdapter(fm) {

    var fragmentList:ArrayList<Fragment> = ArrayList()
    var fragmentListTitles:ArrayList<String> = ArrayList()

    override fun getItem(p0: Int): Fragment {
        return fragmentList.get(p0)
    }

    override fun getCount(): Int {
        return fragmentListTitles.size
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return fragmentListTitles.get(position)
    }

    fun addFragment(fragment: Fragment,title:String){
        this.fragmentList.add(fragment)
        this.fragmentListTitles.add(title)
    }
}